test/models/rule_test.rb

```diff
@@ -41,4 +41,31 @@ class RuleTest < ActiveSupport::TestCase
                .include?(rules(:one)),
            'Expected rule not found by identifier'
   end
+
+  context 'references' do
```

Comment:
The types of URLs we'll get depend on the browser used. For the record, though, `CGI.encode` is _not_ what happens in the browser (Firefox examples shown below, but Chrome seems the same), even though for decoding purposes, `CGI.decode` covers the most ground.

In: `https://ci.foo.redhat.com:1337/api/compliance/rules?&search=reference = 164.308(a)(1)(ii)(B)`
Browser fills: `https://ci.foo.redhat.com:1337/api/compliance/rules?&search=reference%20=%20164.308(a)(1)(ii)(B)`